src/Controller/Action/CreateNewWishlistAction.php

```diff
@@ -0,0 +1,98 @@
+
+namespace BitBag\SyliusWishlistPlugin\Controller\Action;
+
+use BitBag\SyliusWishlistPlugin\Factory\WishlistFactoryInterface;
+use BitBag\SyliusWishlistPlugin\Form\Type\CreateNewWishlistType;
+use Doctrine\Persistence\ObjectManager;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Environment;
+
+final class CreateNewWishlistAction
+{
+    private ObjectManager $wishlistManager;
+
+    private FlashBagInterface $flashBag;
+
+    private TranslatorInterface $translator;
+
+    private UrlGeneratorInterface $urlGenerator;
+
+    private FormFactoryInterface $formFactory;
+
+    private Environment $twigEnvironment;
+
+    private WishlistFactoryInterface $wishlistFactory;
+
+    private TokenStorageInterface $tokenStorage;
+
+    public function __construct(
+        ObjectManager $wishlistManager,
+        FlashBagInterface $flashBag,
+        TranslatorInterface $translator,
+        UrlGeneratorInterface $urlGenerator,
+        FormFactoryInterface $formFactory,
+        Environment $twigEnvironment,
+        WishlistFactoryInterface $wishlistFactory,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->wishlistManager = $wishlistManager;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+        $this->urlGenerator = $urlGenerator;
+        $this->formFactory = $formFactory;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->wishlistFactory = $wishlistFactory;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $token = $this->tokenStorage->getToken();
+        $user = $token ? $token->getUser() : null;
+
+        if ($user instanceof ShopUserInterface) {
+            $wishlist = $this->wishlistFactory->createForUser($user);
+        } else {
+            $wishlist = $this->wishlistFactory->createNew();
+        }
+
+        $form = $this->formFactory->create(CreateNewWishlistType::class, $wishlist);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+
+            $wishlist = $form->getData();
+
+            $this->wishlistManager->persist($wishlist);
+            $this->wishlistManager->flush();
+
+            $this->flashBag->add('success', $this->translator->trans('bitbag_sylius_wishlist_plugin.ui.create_new_wishlist'));
+
+            return new RedirectResponse($this->urlGenerator->generate('bitbag_sylius_wishlist_plugin_shop_wishlist_list_wishlists'));
+        }
```

Comment:
Also, to private method. How about creating an additional command for this?

Reply:
Move all logic to command bus?